bodhi/server/consumers/masher.py

```diff
@@ -588,65 +525,59 @@ def eject_from_mash(self, update, reason):
             update.remove_tag(update.release.pending_testing_tag,
                               koji=buildsys.get_session())
         update.request = None
-        if update.title in self.state['updates']:
-            self.state['updates'].remove(update.title)
-        if update in self.updates:
-            self.updates.remove(update)
         notifications.publish(
             topic="update.eject",
             msg=dict(
                 repo=self.id,
                 update=update,
                 reason=reason,
-                request=self.request,
-                release=self.release,
+                request=self.compose.request,
+                release=self.compose.release,
                 agent=self.agent,
             ),
             force=True,
         )
 
     def init_state(self):
         """
-        Create the mash_dir if it doesn't exist, and generate the lock file path within it.
-
-        This does not create the lock file itself, only the path that will be used for the lock
-        file, which is stored on self.mash_lock.
+        Create the mash_dir if it doesn't exist, and update the Compose state.
 
         Raises:
             Execption: If a lock file already exists and we are not resuming.
```

Comment:
1. This is a typo (old one)
2. I don't think this applies anymore

Reply:
Fixed.